### PR TITLE
Only mask systemd-resolved if WSL is generating resolv.conf

### PR DESCRIPTION
**Pull request by KoviRobi**
I'm using systemd-resolved with tailscale

### CODE REVIEW

File: modules/wsl-distro.nix [hunk 1]
```diff
@@ -108,19 +108,23 @@ with lib; {
...
+                "autovt@".enable = false;
+                firewall.enable = false;
+                # system clock cannot be changed
+                systemd-timesyncd.enable = false;
+                # no udev devices can be attached
+                systemd-udevd.enable = false;
+              }
+              (mkIf config.wsl.wslConf.network.generateResolvConf {
+                systemd-resolved.enable = false;
+              })
```
Comment (SuperSandro2000) [inline modules/wsl-distro.nix:126]:
```suggestion
              systemd-resolved.enable = lib.mkDefault config.wsl.wslConf.network.generateResolvConf;
```
Wouldn't that also work?

Comment (nzbr) [reply modules/wsl-distro.nix:126]: If it does, we wouldn't even need the mkMerge

Comment (KoviRobi) [reply modules/wsl-distro.nix:126]: Apart from the logic error of wanting to negate it to `(!config.wsl.wslConf.network.generateResolvConf)` it does work

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by SuperSandro2000** [comment]
@KoviRobi can you please rebase?

File: modules/wsl-distro.nix [hunk 2]
```diff
@@ -139,7 +139,7 @@ in
             # Disable systemd units that don't make sense on WSL
             services = {
               firewall.enable = false;
-              systemd-resolved.enable = lib.mkDefault false;
+              systemd-resolved.enable = lib.mkDefault (!config.wsl.wslConf.network.generateResolvConf);
```
[the remark this answers is not in the document]
Comment (nzbr) [reply modules/wsl-distro.nix:142]: I actually like the `mkDefault (!...)` more, but it's easier to miss the `!` there when reading it in the future, so I don't have a strong opinion about it

**Comment by K900** [comment]
It's `mkDefault`, your explicit setting should override this already.

**Pull request closed without merging**